Added connector-id flag to kubectl-gs login (#1666)
giantswarm/docs: src/content/use-the-api/management-api/authentication/user/index.md

[![Authentication flow](sso-auth-flow-with-kubectl-gs-login.svg)](sso-auth-flow-with-kubectl-gs-login.svg)

In order to complete the authentication flow, `kubectl gs` will open your default web browser. The page in the browser will prompt you to select an identity provider you want to use for the authentication. If you are authenticated with the identity provider you select (determined e.g. via a cookie), you will see a confirmation page and can close the browser window again. However, if you weren't authenticated with your identity provider yet, you'll have to go through the authentication process you are used to. When this is done, a confirmation page will be shown.

> You can also execute the `login` command with the `--connector-id` flag, and specify an identifier of your preferred identity provider for the authentication:
> 
> ```nohighlight
> kubectl gs login URL --connector-id INDENTITY_PROVIDER_ID
> ```
> 
> When executed with this flag the `login` command does not display the prompt to select an identity provider. Instead, it uses the provider with the specified identifier. In case the provider does not exist, the authentication ends with an error.

Reviewer: I think we should be very careful here with terminology, to keep identity provider and Dex connector apart. With the `--connector-id` flag the user specifies a connector (by its ID), which directly maps to an identity provider.

Another question: As a user, how can I know my connector name, as long as we don't display it in the selection page? I'd add info that there should always be a `customer` connector. And your cluster admin or Giant Swarm support should be able to give out IDs of other connectors.

```suggestion
You can also execute the `login` command with the `--connector-id` flag, and specify an identifier of the desired connector (related to your preferred identity provider) for the authentication:

```nohighlight
kubectl gs login URL --connector-id CONNECTOR_ID
```

When executed with this flag the `login` command does not display the prompt to select an identity provider. Instead, it uses the provider with the specified identifier. In case the provider does not exist, the authentication ends with an error.

**Note:** Connector identifiers are currently not displayed. If you have only one connector configured, it should have the identifier `customer`. To find out identifiers of additional connectors, please contact Giant Swarm support.
```
